### ISSUE TITLE
UI: App state incorrectly shown

### ISSUE THREAD

**Issue by kolloch** [mesosphere/marathon#2568]
An app that I deployed got stuck in delayed state in one UI tab. It doesn't update anymore:

![screen shot 2015-11-05 at 14 24 44](https://cloud.githubusercontent.com/assets/339354/10969496/055ba1ce-83c9-11e5-94cf-f8420773636b.png)

If I open the UI in another tab, it is better:

![screen shot 2015-11-05 at 14 26 34](https://cloud.githubusercontent.com/assets/339354/10969529/40c5bb82-83c9-11e5-9241-c7a86f72fb87.png)

Note that the "Running 1 of 1 tasks" is incorrect, too. There are two tasks.

If I refresh this page, I get:

![screen shot 2015-11-05 at 14 28 32](https://cloud.githubusercontent.com/assets/339354/10969562/869abda6-83c9-11e5-90dd-5b9af74e348d.png)

In the network debug tab of the browser one still sees that the API is continuously queried but the UI doesn't update.

The JSON of the tested app:

```json
{
  "cmd": "python -m SimpleHTTPServer $PORT",
  "cpus": 0.1,
  "disk": 0,
  "healthChecks": [
    {
      "gracePeriodSeconds": 10,
      "ignoreHttp1xx": false,
      "intervalSeconds": 10,
      "maxConsecutiveFailures": 3,
      "path": "/",
      "portIndex": 0,
      "protocol": "HTTP",
      "timeoutSeconds": 3
    }
  ],
  "id": "/test-http",
  "instances": 1,
  "mem": 16,
  "ports": [
    10000
  ]
}
```

The API info got:

```http
HTTP/1.1 200 OK
Cache-Control: no-cache, no-store, must-revalidate
Content-Type: application/json; qs=2
Date: Thu, 05 Nov 2015 13:30:34 GMT
Expires: 0
Pragma: no-cache
Server: Jetty(9.3.2.v20150730)
Transfer-Encoding: chunked
X-Marathon-Leader: http://vbox:8080

{
    "apps": [
        {
            "acceptedResourceRoles": null,
            "args": null,
            "backoffFactor": 1.15,
            "backoffSeconds": 1,
            "cmd": "python -m SimpleHTTPServer $PORT",
            "constraints": [],
            "container": null,
            "cpus": 0.1,
            "dependencies": [],
            "deployments": [],
            "disk": 0,
            "env": {},
            "executor": "",
            "healthChecks": [
                {
                    "gracePeriodSeconds": 10,
                    "ignoreHttp1xx": false,
                    "intervalSeconds": 10,
                    "maxConsecutiveFailures": 3,
                    "path": "/",
                    "portIndex": 0,
                    "protocol": "HTTP",
                    "timeoutSeconds": 3
                }
            ],
            "id": "/test",
            "instances": 2,
            "labels": {},
            "maxLaunchDelaySeconds": 3600,
            "mem": 16,
            "ports": [
                10000
            ],
            "requirePorts": false,
            "storeUrls": [],
            "taskStats": {
                "startedAfterLastScaling": {
                    "stats": {
                        "counts": {
                            "healthy": 1,
                            "running": 1,
                            "staged": 0,
                            "unhealthy": 0
                        },
                        "lifeTime": {
                            "averageSeconds": 690.584,
                            "medianSeconds": 690.584
                        }
                    }
                },
                "totalSummary": {
                    "stats": {
                        "counts": {
                            "healthy": 1,
                            "running": 2,
                            "staged": 0,
                            "unhealthy": 0
                        },
                        "lifeTime": {
                            "averageSeconds": 733.5595,
                            "medianSeconds": 776.535
                        }
                    }
                },
                "withLatestConfig": {
                    "stats": {
                        "counts": {
                            "healthy": 1,
                            "running": 2,
                            "staged": 0,
                            "unhealthy": 0
                        },
                        "lifeTime": {
                            "averageSeconds": 733.5595,
                            "medianSeconds": 776.535
                        }
                    }
                }
            },
            "tasksHealthy": 1,
            "tasksRunning": 2,
            "tasksStaged": 0,
            "tasksUnhealthy": 0,
            "upgradeStrategy": {
                "maximumOverCapacity": 1,
                "minimumHealthCapacity": 1
            },
            "uris": [],
            "user": null,
            "version": "2015-11-05T13:19:03.528Z",
            "versionInfo": {
                "lastConfigChangeAt": "2015-11-05T13:17:37.527Z",
                "lastScalingAt": "2015-11-05T13:19:03.528Z"
            }
        }
    ]
}
```

This was tested on master commit c34dd0aa50c0d0732c798ff20f9b5ddbff5c8fb6.

**Comment by kolloch** [comment]
If I switch back to the index page and then go to the app page again, the information shown is correct.

![screen shot 2015-11-05 at 14 34 05](https://cloud.githubusercontent.com/assets/339354/10969703/4de95a7a-83ca-11e5-848a-92c6bfbec200.png)

If I use browser reload again, it shows the suspended state again.

**Comment by philipnrmn** [comment]
@kolloch this should be the reference implementation for UI bug reports!

I'll take a look.

**Comment by philipnrmn** [comment]
Fixed in mesosphere/marathon-ui#373
